.circleci/config.yml

```diff
@@ -17,11 +17,15 @@ jobs:
       - run:
           name: Push images
           command: |
-            docker push quay.io/$CIRCLE_PROJECT_USERNAME/cluster-api-azure-controller:$CIRCLE_SHA1
+            docker push quay.io/$CIRCLE_PROJECT_USERNAME/cluster-api-azure-controller:${CIRCLE_TAG:-$CIRCLE_SHA1}
 workflows:
   version: 2
   build_and_update:
     jobs:
       - build:
           context:
             - architect
+          # Trigger the job also on git tag.
+          filters:
+            tags:
+              only: /^v[0-9]+\.[0-9]+\.[0-9]+(?:-[0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*)?(?:\+[0-9A-Za-z-]+)?$/
```

Comment:
Why not `/.*/`?

Reply:
container image tags cannot contain `/` for example. (I'm sure there are others like spaces or `:`)

Reply:
because the build of the container fails if the tag name is not a valid semver string

Reply:
I see so it's about tag characters not allowed by the docker registry.